### DISCUSSION TITLE
Any chances of all 3 versions having different signing keys in future?

### DISCUSSION THREAD

**Discussion by NikunjK8** [breezy-weather/breezy-weather#796] [Q&A] [answered]
So i have both Julian and Izzy repo enabled in Droid-ify. I was using the standard version of this app through izzy but recently it got auto updated and replaced with f-droid version through Julian repo since all of them have same signature. It's a bit frustrating. The only way i can prevent this is by either disabling auto updates and check the version manually everytime or disabling julian repo, i can't do both for obvious reasons. So is there any plan of all version having different keys which would solve the problem? Or any other way to prevent this from happening which I'm not aware of?

**Comment by papjul** [comment]
Having the same signing key is an intended feature:
- allows users to easily try different flavors.
- different signing keys/applicationId requires an import/export feature and it's a bit complex to implement
- we plan on publishing the app to the default F-Droid repository, and this repo is lagging behind in updates, and we want users to easily update to latest version before they report an issue on GitHub.

However, I understand your issue given how repos work. Maybe after app is added to F-Droid default repo, we could ask Izzy and Julian to remove the app from their repos and only keep 3 Breezy Weather repos (one for each flavor and the user decides which repo flavor they want)?

**Reply by NikunjK8** [reply]
Yeah that's actually a good solution. Make sure to inform users through release page when repos are ready.

**Reply by papjul** [reply]
Julian removed Breezy Weather. The F-Droid flavor is now available from our own repos. I will leave it in Izzy repo until it lands in F-Droid default repo.

**Reply by NikunjK8** [reply]
Hi! If i remember correctly f-droid sign every app with their own signature keys right?

**Reply by papjul** [reply]
Yes and no, the recommended way for new apps is to sign it with the developer key, much easier to update from a different store: https://f-droid.org/fr/docs/Reproducible_Builds/

**Reply by NikunjK8** [reply]
I see, unfortunately my problem still remains then. I'm on standard version and it can get replaced with f-droid version when f-droid build gets released in their repo. I think I'll have to use another method for updates.

**Reply by papjul** [reply, marked as answer]
F-Droid default repo updates about once a week. Unless your auto-update only happens about once a week, I don't think it can happen.

Let's say:
1) we release v5.1.6 on 2024-03-17
2) you auto-update on 2024-03-18
3) F-Droid default repo updates on 2024-03-24

Since it's the same version code, there won't be another auto update. It's even possible I release v5.1.7 between 2024-03-18 and 2024-04-24 while F-Droid default repo will still be on v5.1.6

Let me know if I missed something?

**Reply by NikunjK8** [reply]
Yeah i completely forgot f-droid releases are always delayed compared to others. Thanks for clarifying. One last question, does all the settings remain when you switch between all 3 versions?

**Reply by papjul** [reply]
Yes, but some settings may temporarily change when going from Standard/Gplay flavors to F-Droid flavor, since that one excludes non-free networks.

For sources on each individual location, you may encounter a "Source no longer available" message if you added a non-free network source on Standard/Gplay then switched to F-Droid. A troubleshooting was added to explain what to do.

Some settings may fallback to default value, such as "default weather source for new locations" or "location source", or "location search source".

As long as you don't manually change these settings, going back to Standard/Gplay will bring back settings as they were.

**Reply by NikunjK8** [reply]
Got it. Thanks for your time. ❤️
